lib/xnvme_be_vfio_mem.c

```diff
@@ -71,6 +71,44 @@ xnvme_be_vfio_buf_vtophys(const struct xnvme_dev *XNVME_UNUSED(dev), void *XNVME
 	errno = ENOSYS;
 	return -ENOSYS;
 }
+
+int
+xnvme_be_vfio_mem_map(const struct xnvme_dev *dev, void *vaddr, size_t nbytes, uint64_t *phys)
+{
+	struct xnvme_be_vfio_state *state = (void *)dev->be.state;
+	struct vfio_container *vfio = state->ctrl->pci.dev.vfio;
+	int err;
+
+	XNVME_DEBUG("xnvme_be_vfio_mem_map(%p, %p)", dev, vaddr);
+
+	if (!ALIGNED(nbytes, __VFN_PAGESIZE)) {
```

Comment:
You should also check `vaddr` actually, i.e.

    if (!ALIGNED(vaddr | nbytes, __VFN_PAGESIZE)) { ...